### PR TITLE
Dev/sol/refactor ii

### PR DESCRIPTION
**Pull request by SoloJacobs**

### CODE REVIEW

File: v2/robotmk/src/environment.rs [hunk 1]
```diff
@@ -65,16 +64,20 @@ fn build_environment<'a>(
             let drop_suite = matches!(environment_build_status, EnvironmentBuildStatus::Failure(_));
             environment_build_states_administrator
                 .insert_and_write_atomic(&suite.name, environment_build_status)?;
-            drop_suite
+            if drop_suite {
```
Comment (jherbel) [inline v2/robotmk/src/environment.rs:67]: I think we have to invert this, right? We want to return `Some(suite)` if we the building suceeded.

File: v2/robotmk/src/environment.rs [hunk 2]
```diff
@@ -65,16 +64,20 @@ fn build_environment<'a>(
...
+            } else {
+                None
+            }
         }
         None => {
             debug!("Nothing to do for suite {}", suite.name);
             environment_build_states_administrator
                 .insert_and_write_atomic(&suite.name, EnvironmentBuildStatus::NotNeeded)?;
-            false
+            None
```
Comment (jherbel) [inline v2/robotmk/src/environment.rs:77]: See above

**Review by jherbel — APPROVED** [review]
Nice idea :+1: Just a small comment regarding the discard logic.

**Review by jherbel — CHANGES_REQUESTED** [review]
Sorry, chose "Approve" instead of "Request changes" before

File: v2/robotmk/src/environment.rs (hunk 1 above)
Comment (SoloJacobs) [reply v2/robotmk/src/environment.rs]: Yes, good catch. Not sure how to test this :thinking:

File: v2/robotmk/src/environment.rs (hunk 2 above)
Comment (SoloJacobs) [reply v2/robotmk/src/environment.rs]: Done

**Review by jherbel — APPROVED** [review]

**Pull request merged**